sklearn/linear_model/tests/test_ridge.py

```diff
@@ -896,6 +896,8 @@ def test_ridge_gcv_sample_weights(
     ridge_reg = Ridge(alpha=kfold.alpha_, fit_intercept=fit_intercept)
     splits = cv.split(X_tiled, y_tiled, groups=indices)
     predictions = cross_val_predict(ridge_reg, X_tiled, y_tiled, cv=splits)
+    if predictions.shape != y_tiled.shape:
+        predictions = predictions.reshape(y_tiled.shape)
```

Comment:
It's a bit strange the `cross_val_predict` does not have the same shape as `y[indices]`.

I'm guessing it's because `Ridge.predict` will now always return an ndarray of `(n_samples,)`. If `Ridge` was trained on a `y` of shape `(n_samples, 1)`, should `Ridge.predict` return a prediction of shape `(n_samples, 1)`?

Reply:
We have in other places where things fail if the shape is `(n_samples, 1)`. I'm more in favor of having a consistent output shape for single outputs. We probably want `cross_val_predict` to become more consistent with the rest of the library.